Review-code

mre opened the pull request.

mre committed:
How To Review Code

mre committed:
line breaks

mre committed:
wording

mre committed:
wording

mre committed:
wording

mre committed:
Update "How To Review Code" article: revise date, adjust tags, and enhance content clarity

mre committed:
Pretty screenshots

mre committed:
wording

mre committed:
wooording

M3t0r commented on content/2025/how-to-review-code/index.md:
```diff
@@ -0,0 +1,307 @@
+That's the purpose of the review process: there is no guarantee that the code will be accepted. 
+
+In open source, many people will contribute code that doesn't meet your standards.
+There needs to be someone who says "no" and this is a very unpopular job (ask Linus Torvalds).
```
I'd not mention Linus. He isn't unpopular because he says no, he's unpopular because of the way he says no. Keep it generic like "ask any successful open source maintainer" or drop the whole parenthesis.

mre replied:
okay, I went with "open source maintainer." 👍  I dropped the "successful" bit, as it's subjective and not relevant for my point.

M3t0r commented on content/2025/how-to-review-code/index.md:
```diff
@@ -0,0 +1,307 @@
+However, great projects need gatekeepers because the alternative is subpar code and eventually unmaintainable projects. 
+
+At times, people will say "let's just merge this and fix it later."
+That's a slippery slope.
```
```suggestion
I think that's a slippery slope.
```

M3t0r commented on content/2025/how-to-review-code/index.md:
```diff
@@ -0,0 +1,307 @@
+
+## Don't Be A Gatekeeper
+
+Don't be a gatekeeper who only accepts code that meets your personal standards.
```
You just said the complete opposite in the paragraph above. I _think_ the difference you see is "your standards" and "the project standard". Especially when starting out they are the same in projects though. And you never said "don't be afraid to say no if the proposed changes are not meeting the _projects_ standards".

mre replied:
Okay, I merged it with the previous section. It now reads:


> ## Don't Be Afraid To Say "No"
> 
> ...
> 
> When it gets hard, remember that you're not rejecting the person, you're rejecting the code.
> Remind people that you appreciate their effort and that you want to help them improve.
> 
> **Even though you'll develop an intuition for what to focus on in reviews, you should still back it up with facts.
> If you find yourself saying "no" to the same thing over and over again, consider writing a style guide or a set of guidelines for your team.**
> 
> Be gracious but decisive.
> It's just code.

M3t0r commented on content/2025/how-to-review-code/index.md:
```diff
@@ -0,0 +1,307 @@
+Be gracious but decisive.
+It's just code.
+
+## Don't Be A Gatekeeper
```
One could argue that Linus Torvalds job is exclusively being a gatekeeper.

And in the first sentence of this chapter you already qualify: "i didn't mean you shouldn't be a gatekeeper, i meant you shouldn't be _this kind_ of gatekeeper".

The headline is to binary/black and white for the content you put under it.

mre replied:
removed that section

M3t0r commented on content/2025/how-to-review-code/index.md:
```diff
@@ -0,0 +1,307 @@
+Otherwise, what's the point of a code review in the first place?
+That's a mindset shift that's important to make.
+
+Reviews aren't about pointing out flaws, they're about creating a shared understanding of the code within the team.
```
Yes, they _are_ about pointing out flaws. One might say that is the most fundamental purpose. You want to say they are not _exclusively_ about pointing out flaws.

mre replied:
yup, changed.

M3t0r commented on content/2025/how-to-review-code/index.md:
```diff
@@ -0,0 +1,307 @@
+## Be Upfront About Your Availability 
+
+Code reviews are often a bottleneck in the development process,
+because they can't be automated.
```
Yes they can. That's what CI is. _Parts_ of it can't be automated. The human parts of understanding and questioning.

mre replied:
Yes!

Rephrased it to 

> Code reviews are often a bottleneck in the development process, because they can't be fully automated:
there's a human in the loop who has to look at the code and provide feedback.

M3t0r commented on content/2025/how-to-review-code/index.md:
```diff
@@ -0,0 +1,307 @@
+When reviewing code, focus on the reasoning behind the changes.
+This has a much better chance of success than pointing out flaws without reasoning. 
+
+![](bad-review-comment.png)
```
This picture has no connection to the paragraph before it. It's a change in medium and in topic. I thought I didn't get the connection and reread the paragraph before moving on and figuring out that they are not connected.

mre replied:
Hum, maybe it's clearer when looking at the rendered version?

mre replied:
I added some context. Still not 100% happy, because of accessibility issues, but I think it's better now than before.

M3t0r commented on content/2025/how-to-review-code/index.md:
```diff
@@ -0,0 +1,307 @@
+![](good-review-comment.png)
+
+I realize that this requires more time and effort, but it's worth it!
+Most of the time, the author will appreciate it and avoid making the same mistake in the future, s there is a compound effect from helpful reviews over time.
```
```suggestion
Most of the time, the author will appreciate it and avoid making the same mistake in the future, there is a compound effect from helpful reviews over time.
```

mre replied:
fixed

M3t0r reviewed (changes requested):
Overall the sections feel disconnected, but worse sometime contradictory ('Don't Be Afraid To Say "No"' vs 'Don't Be A Gatekeeper').

> This blog post is a summary of my current approach to code reviews.

"Summary" makes me think that this is written in one go, and is not a somewhat loose collection of notes and tips you collected. This sets the wrong expectations from the post (which you don't fulfil, and probably don't want to).

mre committed:
intro wording

mre committed:
update date

mre committed:
bye bye Linus

mre committed:
accept suggestion

mre committed:
merge sections

mre committed:
> Reviews aren't *exclusively* about pointing out flaws, they're also about creating a shared understanding of the code within the team.
I often learn the most about writing better code by reviewing other people's code.
I've also gotten excellent feedback on my own code from excellent engineers.

mre committed:
clarify the human in the loop

mre committed:
wording fixes

mre committed:
Mention Simon as reviewer

M3t0r reviewed (approved):
Looks good! Send it 🚀

mre committed:
Time flies
